core/test/utils/batch_test_utils.hpp

```diff
@@ -316,7 +316,7 @@ void test_solve(std::shared_ptr<const Executor> exec, const size_t nbatch,
     auto right_scale = Dense::create(exec, s_vec_sz);
```

Comment:
Is left_scale always equal to right_scale?

Reply:
No, not always. They can be different. Here in this test I just use the same values. It does not matter here because we're not actually testing the final solution; only the final residual.